index.d.ts

```diff
@@ -0,0 +1,78 @@
+import { Action, StoreEnhancer, Store } from 'redux';
+
+interface StoreCreator {
+  <S, T>(
+    reducer: EnhancedReducer<S, T>,
+    preloadedState: S,
+    enhancer: StoreEnhancer<S>
+  ): Store<S>;
+}
+
+interface EnhancedReducer<S, T> {
```

Comment:
the reducer type in redux is just a function type.

`export type Reducer<S> = (state: S, action: AnyAction) => S;`

can this not be one also?

also, I think we know that T is either an S or a loop(S, CmdType) right? Can we restrict it to that?

Reply:
ah i see you can user interfaces the same way as the type itself. but the last question still applies

Reply:
One instance where `EnhancedReducer` gets a different return type via the generic is in `combineReducers` where we know that the response is always going to be a `loop(S, CmdType)` and never just State:

```
declare function combineReducers<S, T>(
  reducers: StrictReducerMapObject<T>
): EnhancedReducer<S, [S, CmdType<Action, Function>]>;
```

Not sure if that's worth adding an extra generic for though? Maybe there's a more concise way of expressing that in this case we know which member of the union type will be returned.